dash/dash.py

```diff
@@ -131,7 +130,13 @@ def __init__(
         self._assets_url_path = assets_url_path
 
         # allow users to supply their own flask server
-        self.server = server or Flask(name, static_folder=static_folder)
+        if server:
+            if isinstance(server, Flask):
+                self.server = server
+            else:
+                self.server = Flask(name, static_folder=static_folder)
+        else:
+            self.server = None
```

Comment:
I like the choice of `True` (auto), `False` (defer) and `Flask(...)` (use existing) but then the default value should be updated to `True`. I'd also say, given that we're now accepting multiple value types for `server`, that we should explicitly throw an error if we get neither a `Flask` nor a `bool` - I know @chriddyp likes making specific types in `exceptions.py`, but I'd be fine with just something like `ValueError`

So something like:
```py
if isinstance(server, bool):
    self.server = Flask(...) if server else None
elif isinstance(server, Flask):
    self.server = server
else:
    raise ValueError('server must be a Flask app or a boolean')
```